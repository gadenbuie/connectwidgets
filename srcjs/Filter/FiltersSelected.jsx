import PropTypes from 'prop-types';
import { FilterContext } from './stateUtils';
import './filtersSelected.scss';

function FiltersSelected({ type }) {
  const context = React.useContext(FilterContext);
  const selections = context.stateSelectionsFor(type);

  const onDiscard = selection => {
    context.updateSelection({
      ...selection,
      isSelected: false,
    });
  }

  return (
    <div className="rscfilter-selections">
      {
        selections.map(selection => (
          <div
            key={selection.name}
            className="rscfilter-selections__item badge rounded-pill bg-secondary"
          >
            { selection.label }
            {/* eslint-disable */}
            <span
              className="rscfilter-selections__item-discard"
              onClick={() => onDiscard(selection)}
              role="button"
            />
            {/* eslint-enable */}
          </div>
        ))
      }      
    </div>
  );
}

FiltersSelected.propTypes = {
  type: PropTypes.string,
};

export default FiltersSelected;
