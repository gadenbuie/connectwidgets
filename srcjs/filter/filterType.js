/**
 * Filter types
 */
 export const FilterType = {
  owner: 'owner',
  type: 'type',
  tag: 'tag',
};
