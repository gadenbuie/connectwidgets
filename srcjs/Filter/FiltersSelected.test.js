import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FiltersSelected from './FiltersSelected';
import { FilterType } from './filterType';

const data = [
  {
    name: 'API',
    label: 'API',
    isSelected: true,
  },
  {
    name: 'Application',
    label: 'Application',
    isSelected: true,
  },
  {
    name: 'Document',
    label: 'Document',
    isSelected: true,
  },
];
const fakeContext = {
  selections: { [FilterType.type]: data },
  stateSelectionsFor: jest.fn().mockImplementation(() => data),
  updateSelection: jest.fn(),
};
jest.spyOn(React, 'useContext').mockReturnValue(fakeContext);

const mkWrapper = () => shallow(<FiltersSelected type={FilterType.type} />);

describe('<FiltersSelected />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  it('renders properly', () => {
    const wrapper = mkWrapper();
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('.rscfilter-selections__item').length).toBe(3);
  });
  
  it('emits changes to context/state', () => {
    const wrapper = mkWrapper();
    const selections = wrapper.find('.rscfilter-selections__item');
    
    expect(fakeContext.updateSelection).not.toHaveBeenCalled();
    selections.at(1).find('.rscfilter-selections__item-discard').simulate('click', {});
    expect(fakeContext.updateSelection).toHaveBeenCalledWith({
      name: 'Application',
      label: 'Application',
      isSelected: false,
    });

    selections.at(2).find('.rscfilter-selections__item-discard').simulate('click', {});
    expect(fakeContext.updateSelection).toHaveBeenCalledWith({
      name: 'Document',
      label: 'Document',
      isSelected: false,
    });
  });
});
