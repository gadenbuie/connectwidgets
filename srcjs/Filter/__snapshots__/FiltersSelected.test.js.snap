// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FiltersSelected /> renders properly 1`] = `
<div
  className="rscfilter-selections"
>
  <div
    className="rscfilter-selections__item badge rounded-pill bg-secondary"
    key="API"
  >
    API
    <span
      className="rscfilter-selections__item-discard"
      onClick={[Function]}
      role="button"
    />
  </div>
  <div
    className="rscfilter-selections__item badge rounded-pill bg-secondary"
    key="Application"
  >
    Application
    <span
      className="rscfilter-selections__item-discard"
      onClick={[Function]}
      role="button"
    />
  </div>
  <div
    className="rscfilter-selections__item badge rounded-pill bg-secondary"
    key="Document"
  >
    Document
    <span
      className="rscfilter-selections__item-discard"
      onClick={[Function]}
      role="button"
    />
  </div>
</div>
`;
